app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -121,7 +167,38 @@ class StateFragmentPresenter @Inject constructor(
 
   private fun subscribeToCurrentState() {
     ephemeralStateLiveData.observe(fragment, Observer<EphemeralState> { result ->
-      logger.d("StateFragment", "getCurrentState: ${result.state.name}")
+      if (result.hasState()) {
+        itemList.clear()
+        currentEphemeralState.set(result)
+
+        updateDummyStateName()
+
+        val interactionId = result.state.interaction.id
+        val hasPreviousState = result.hasPreviousState
+        var hasNextState = false
+        hasGeneralContinueButton = false
+
+        if (!result.terminalState) {
```

Comment:
@BenHenning  What do you mean by this?

Reply:
I'm referring to the stateTypeCase you're checking below. Let's use that here to check the type, as well. stateTypeCase will be one of PENDING/COMPLETED/TERMINAL. The terminal boolean is only true if the state is terminal, but that's just because I needed to put something there in the oneof case in the proto for it to be populated correctly (https://github.com/oppia/oppia-android/blob/develop/model/src/main/proto/exploration.proto#L165). We should use stateTypeCase to determine the states and not the constituent values that each case corresponds to.

Reply:
Okay ack. Will update code accordingly.

Reply:
Changed condition to `result.stateTypeCase != EphemeralState.StateTypeCase.TERMINAL_STATE`